core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/impl/PoolImpl.java

```diff
@@ -383,12 +386,20 @@ public Group getWatchGroup() {
   @Override
   public boolean hasDuplicateIndices() {
     Set<String> indices = new HashSet<>();
+    Set<String> duplicateSequences = new HashSet<>();
     for (PoolableElementView item : getPoolableElementViews()) {
       if (hasDuplicateIndices(indices, item)) {
-        return true;
+        for (Index index : item.getIndices()) {
+          duplicateSequences.add(index.getSequence());
+        }
       }
     }
-    return false;
+    if (duplicateSequences.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)